python/pylibcudf/pylibcudf/contiguous_split.pyx

```diff
@@ -111,6 +127,178 @@ cdef class PackedColumns:
         )
 
 
+cdef class ChunkedPack:
+    """
+    A chunked version of :func:`pack`.
+
+    This object can be used to pack (and therefore serialize) a table
+    piece-by-piece through a user-provided staging buffer. This is
+    useful when we want the end result to end up in host memory, but
+    want control over the memory footprint.
+    """
+    def __init__(self):
+        raise ValueError(
+            "ChunkedPack should not be constructed directly. Use create instead."
+        )
+
+    @staticmethod
+    def create(
+        Table input,
+        size_t user_buffer_size,
+        Stream stream,
+        DeviceMemoryResource temp_mr,
+    ):
+        """
+        Create a chunked packer.
+
+        Parameters
+        ----------
+        input
+            The table to pack.
+        user_buffer_size
+            Size of the staging buffer to pack into, must be at least 1MB.
```

Comment:
I'm assuming this raises in libcudf?

Reply:
Yes it does https://github.com/rapidsai/cudf/blob/branch-25.06/cpp/src/copying/contiguous_split.cu#L2102